Tests/Legacy/Shopware/Connect/ProductToShopTest.php

```diff
@@ -524,12 +529,27 @@ public function testInsertArticleAndAutomaticallyCreateCategories()
             $childCategoryModel->getParent()->getName()
         );
 
+        $categoryRepository = Shopware()->Models()->getRepository('Shopware\Models\Category\Category');
+        /** @var \Shopware\Models\Category\Category $childCategoryModel */
+        $childCategoryModel2 = $categoryRepository->findOneBy(['name' => $childCategory2]);
+        $this->assertInstanceOf('Shopware\Models\Category\Category', $childCategoryModel2);
+        $this->assertEquals(
+            $childCategoryModel->getName(),
+            $childCategoryModel2->getParent()->getName()
+        );
+        $this->assertEquals(
+            $childCategory2,
+            $childCategoryModel2->getName()
+        );
+
         /** @var Article $article */
         $article = $this->modelManager->getRepository(Article::class)->findOneByName($product->title);
 
         $assignCategories = $article->getCategories();
-        $this->assertEquals(1, count($assignCategories));
-        $this->assertEquals($childCategory, $assignCategories[0]->getName());
+        $this->assertEquals(2, count($assignCategories));
+        //this category is assigned to the product when it is created
+        $this->assertEquals('Bücher', $assignCategories[0]->getName());
```

Comment:
why? Probably, because Bücher was mapped to local category in previous test ?

Reply:
no it's hard coded in $this->getProduct()
i just overwrite the categories instead of merge them